[dasha3412] iP

dasha3412 opened the pull request:
# Victor TaskBot
The Victor TaskBot can help you keep track of:
* To-Dos;
* Deadlines; and
* Events

Enter a prompt as simple as "todo Buy a Car" and you will see:
>   ~  Cool! I added this task:
>
>   ~  {T}{ } Buy a Car
>
>   ~  You now have 1 task in your list.

**How to get started:**
1. Download the latest release from [here](https://github.com/dasha3412/ip/releases)
2. Move the victor.jar file to a directory of your choice (called dir_name in next steps)
3. Open your terminal and move to dir_name
4. Run the following command: java -jar victor.jar
5. Track your tasks!

You should see the following if everything runs correctly:
```
,---.  ,---..-./`)     _______ ,---------.    ,-----.    .-------.
|   /  |   |\ .-.')   /   __  \\          \ .'  .-,  '.  |  _ _   \    
|  |   |  .'/ `-' \  | ,_/  \__)`--.  ,---'/ ,-.|  \ _ \ | ( ' )  |
|  | _ |  |  `-'`"`,-./  )         |   \  ;  \  '_ /  | :|(_ o _) /
|  _( )_  |  .---. \  '_ '`)       :_ _:  |  _`,/ \ _/  || (_,_).' __
\ (_ o._) /  |   |  > (_)  )  __   (_I_)  : (  '\_/ \   ;|  |\ \  |  |
 \ (_,_) /   |   | (  .  .-'_/  ) (_(=)_)  \ `"/  \  ) / |  | \ `'   /
  \     /    |   |  `-'`-'     /   (_I_)    '. \_/``".'  |  |  \    /
   `---`     '---'    `._____.'    '---'      '-----'    ''-'   `'-'

Hello! My name is Victor!
What can I do for you?
============================================================
```

All the commands the Victor TaskBot can handle are handled by this method:
```java
public Command parseInput(String input) {
        Command command;
        String[] inputWords = input.trim().split(" ");
        command = switch (inputWords[0]) {
        case "todo" -> new ToDoCommand(inputWords);
        case "deadline" -> new DeadlineCommand(inputWords);
        case "event" -> new EventCommand(inputWords);
        case "delete" -> new DeleteCommand(inputWords);
        case "mark" -> new MarkCommand(inputWords);
        case "unmark" -> new UnmarkCommand(inputWords);
        case "list" -> new ListCommand(inputWords);
        case "bye" -> new ExitCommand(inputWords);
        case "find" -> new FindCommand(inputWords);
        default -> new Command(inputWords);
        };
        return command;
    }
```

Get started with Victor TaskBot to help you track all your To-Dos, Deadlines, and Events today 🥰 

Check off what features you make use of as you go!
- [ ] To-Dos
- [ ] Deadlines
- [ ] Events

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

dasha3412 committed:
Changing name and adding introduction

dasha3412 committed:
Able to take user input and echo back, exits upon bye command

dasha3412 committed:
Saving input and listing

dasha3412 committed:
Fixing lines

dasha3412 committed:
Making task list

dasha3412 committed:
Split task from Victor

dasha3412 committed:
Adding support for To Dos, Events, and Deadlines

dasha3412 committed:
Making automated test scripts

dasha3412 committed:
Adding some error handling and personality additions

dasha3412 committed:
Adding delete functionality

dasha3412 committed:
Change boolean variable names

dasha3412 committed:
Make class member fields protected

dasha3412 committed:
Fix line widths to meet Java standards

dasha3412 committed:
Fix search for keywords to take exact words with no extra chars

dasha3412 committed:
Split logical units with spacing and fix delete message

dasha3412 committed:
Update README and remove randomisation for task descriptions

dasha3412 committed:
Update README

dasha3412 committed:
Add write to file functionality

dasha3412 committed:
Merge branch 'branch-Level-7'
Merge level 7 functionality to master branch.

dasha3412 committed:
Add date-time processing

dasha3412 committed:
Merge branch 'branch-Level-8'

dasha3412 committed:
Make system loading OOP-style

dasha3412 committed:
Add modified files from OOP work

dasha3412 committed:
Make UI, storage, parse, tasks, commands fully OOP

dasha3412 committed:
Merge branch 'branch-A-MoreOOP'

dasha3412 committed:
Reorganise files into common package

dasha3412 committed:
Merge branch 'branch-A-Packages'

dasha3412 committed:
Change gitinore to ignore data

dasha3412 committed:
Merge branch 'add-gradle-support'

dasha3412 committed:
Use Gradle to automate functions

dasha3412 committed:
Add tests for parser and tasklist

dasha3412 committed:
Fix formatting and JAR file name

dasha3412 committed:
Add JavaDoc comments for methods except in tasks

dasha3412 committed:
Fix to match coding standards

dasha3412 committed:
Add find functionality

dasha3412 committed:
Merge branch 'A-CodingStandard'

dasha3412 committed:
Fix merge conflicts from JavaDoc branch

dasha3412 committed:
Fix merge conflicts from level 9 branch

dasha3412 committed:
Add more JavaDoc comments

teamotitan commented on src/main/java/victor/Main.java:
```diff
@@ -0,0 +1,67 @@
+        ui = new UI();
+        storage = new Storage(filePath);
+        taskList = storage.load();
+        ui.showWelcomeMessage();
```
I like that the method clearly explained what it is going to do

teamotitan commented on src/main/java/victor/Main.java:
```diff
@@ -0,0 +1,67 @@
+    /**
+     * Runs a program loop that continually received user input until the UI receives a "bye" command.
+     */
+    private void runUntilExit() {
```
Again, I like that the method is very clear in saying what it is going to do

teamotitan commented on src/main/java/victor/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,92 @@
+    }
+
+    public String[] getAdditionalInput() {
+        return this.additionalInput;
```
I like that you have been using verbs for actions

teamotitan commented on src/main/java/victor/tasklist/TaskList.java:
```diff
@@ -0,0 +1,155 @@
+
+import victor.tasks.Task;
+
+public class TaskList {
```
I see the usage of nouns for classes. Nice!

teamotitan reviewed (commented):
LGTM!

dasha3412 committed:
Fix checkstyle violations

dasha3412 committed:
Adapt program structure into GUI version

dasha3412 committed:
Fix scrolling to bottom for GUI

dasha3412 committed:
Make essential GUI features

dasha3412 committed:
Merge branch 'branch-Level-10'

dasha3412 committed:
Edit handler javadoc comment

dasha3412 committed:
Comment where varargs is used

LimKaiWei commented on src/main/java/victor/messages/ReturnMessage.java:
```diff
@@ -0,0 +1,44 @@
+package victor.messages;
```
i like how you used this class to wrap return messages

LimKaiWei commented on src/main/java/victor/Main.java:
```diff
@@ -0,0 +1,36 @@
+ */
+public class Main extends Application {
+    /**
+     * Called by the Launcher main method.
```
should there be a `@param` for this comment?

LimKaiWei commented on src/main/java/victor/Main.java:
```diff
@@ -0,0 +1,36 @@
+            fxmlLoader.<MainWindow>getController().welcomeUser();
+            stage.show();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Is it better to handle this more gracefully instead of printing stacktrace?

LimKaiWei commented on src/test/java/src/Main.java:
```diff
@@ -0,0 +1,6 @@
+/** Arbitrary main class for src file of test folder */
```
Is this file needed?

LimKaiWei commented on src/test/java/victor/parser/ParserTest.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    @Test
+    public void parseNoNameDeadlineTest() {
```
i like how you include cases for non conforming user inputs

dasha3412 committed:
Add java assertions to check values

Values are not being checked before and after operations.

Some relationships assumed to be true may not hold.

Let's add java assertions to check that values change correctly
after applying operations, especially on ArrayLists.

Assertions are effective to check value relationships as they
can be enabled or disabled without changing code functionality.

Assertions added mostly to check size of tasks ArrayList and
verify that additional is provided for some commands like Delete.

dasha3412 committed:
Refactor code to match code quality guidelines

Code quality issues like nested conditionals and
duplicated code impede readability.

Difficult to read and understand code easily.

Let's do code refactoring by extracting duplicated code as methods,
long expressions calls as methods, introducing guard clauses, etc. to
improve code readability.

Refactoring code will make it more easily understood and avoid repetitions
so that other programmers and people auditing code can understand logical flow
better and more easily.

Refactoring done following guidelines found at https://refactoring.com/catalog/

dasha3412 committed:
Use streams for filtering when finding tasks

dasha3412 committed:
Merge pull request #1 from dasha3412/branch-A-Assertions

Add java assertions to check values

dasha3412 committed:
Fix merge conflicts

dasha3412 committed:
Merge branch 'master' into branch-A-Streams

dasha3412 committed:
Merge pull request #2 from dasha3412/branch-A-CodeQuality

Refactor code to match code quality guidelines

dasha3412 committed:
Merge branch 'master' into branch-A-Streams

dasha3412 committed:
Merge pull request #3 from dasha3412/branch-A-Streams

Use streams for filtering when finding tasks

dasha3412 committed:
Set up continuous integration

dasha3412 committed:
Try to fix yml build

dasha3412 committed:
Test git push for build

dasha3412 committed:
Check contents of directory

dasha3412 committed:
Fix runtest to pass CI tests

dasha3412 committed:
Fix runtest to pass CI tests, add files

dasha3412 committed:
Try apt-get dos2unix

dasha3412 committed:
Update apt-get to try install dos2unix

dasha3412 committed:
Change input file in runtime.sh

dasha3412 committed:
Remove IO redirection

dasha3412 committed:
Add detecting duplicates functionality

dasha3412 committed:
Merge branch 'branch-BCD-Extension'

dasha3412 committed:
Make GUI better (more coquette)

dasha3412 committed:
Merge branch 'branch-A-BetterGui'

dasha3412 committed:
Change personality in some messages

dasha3412 committed:
Add exit case and handle exit in main window

dasha3412 committed:
Merge branch 'branch-A-MoreErrorHandling'

dasha3412 committed:
Add more unit tests and fix file write

dasha3412 committed:
Merge branch 'branch-A-MoreTesting'

dasha3412 committed:
Add error handling for wrong text in data file

dasha3412 committed:
Add UI image and title for taskbot window

dasha3412 committed:
Add user guide and update UI image

dasha3412 committed:
Update readme

dasha3412 committed:
Update main project README

dasha3412 committed:
Make code adhere to SLAP
